doc/source/whatsnew/v1.3.0.rst

```diff
@@ -753,7 +753,7 @@ Plotting
 - Prevent warnings when matplotlib's ``constrained_layout`` is enabled (:issue:`25261`)
 - Bug in :func:`DataFrame.plot` was showing the wrong colors in the legend if the function was called repeatedly and some calls used ``yerr`` while others didn't (partial fix of :issue:`39522`)
 - Bug in :func:`DataFrame.plot` was showing the wrong colors in the legend if the function was called repeatedly and some calls used ``secondary_y`` and others use ``legend=False`` (:issue:`40044`)
-
+- Bug in :func:`DataFrame.plot` was taking ``xlim`` and ``ylim`` kwargs even if they had string elements, now ``ValueError`` is raised if elements are not ``int`` or ``float`` dtype (:issue:`40781`)
```

Comment:
I _think_ matplotlib could also take datetime object, e.g. something like `ax.set_xlim([datetime1, datetime2])`, could you please check if it is case? If so, would be nice to make changes accordingly in the PR to reflect.

Reply:
I added datetime datatype to the filter too, I would love a feedback on the function.

Reply:
make this simpler like was not raising for invalid dtyped elements to xlim/ylim